src/Sulu/Bundle/PageBundle/Trash/PageTrashItemHandler.php

```diff
@@ -183,7 +183,7 @@ public function restore(TrashItemInterface $trashItem, array $restoreFormData =
             $localizedPage->setStructureType($localeData['structureType']);
             $localizedPage->getStructure()->bind($localeData['structureData']);
             $localizedPage->setExtensionsData($localeData['extensionsData']);
-            $localizedPage->setPermissions($localeData['permissions']);
+            $localizedPage->setPermissions($localeData['permissions'] ?: []);
```

Comment:
this will fix restore for elements which were stored with permissions = null. We currently have no way to reproduce how it happens but it seems like old sulu which did upgrade are sometimes running into this case as it was reported by two different sulu users.